hedgehog-example/test/Test/Example/STLC.hs

```diff
@@ -0,0 +1,281 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+module Test.Example.STLC where
+
+
+import           Control.Applicative
+import           Control.Monad
+import           Control.Monad.Morph
+import           Control.Monad.Reader
+
+import           Data.Either
+import           Data.Map.Strict (Map)
+import qualified Data.Map.Strict as M
+import           Data.Maybe
+import           Data.Monoid
+import           Data.Set (Set)
+import qualified Data.Set as S
+import           Data.Text (Text)
+
+import           Hedgehog
+import qualified Hedgehog.Gen as HG
+import qualified Hedgehog.Range as HR
+
+import           Prelude
+
+
+-- -----------------------------------------------------------------------------
+-- A simply-typed lambda calculus with ints, bools, and strings
+
+data Type =
+    TBool
+  | TInt
+  | TString
+  | TArrow Type Type
+  deriving (Eq, Ord, Show)
+
+data Expr =
+    EBool Bool
+  | EInt Int
+  | EString Text
+  | EVar Text
+  | ELam Text Type Expr
+  | EApp Expr Expr
+  deriving (Eq, Ord, Show)
+
+-- -----------------------------------------------------------------------------
+
+-- | Evaluate to weak head normal form.
```

Comment:
would probably get more interesting failures if this went all the way to normal form, was just being lazy.